src/api/roles/coach/coach.py

```
@@ -65,9 +66,83 @@ def create_coach_request(coach_details: CoachRequestInput, db = Depends(get_sess
 
     return CreateCoachRequestResponse(coach_request_id=cr.id, coach_id=coach.id) # type: ignore
 
+# Updating coach request , which includes coach availability, experience, and certifications.
+@router.patch("/update_coach_info", response_model=UpdateCoachInfoResponse)
+def update_coach_request(new_coach_details: UpdateCoachInfoInput, db = Depends(get_session), coach_acc: Coach = Depends(get_coach_account)):
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)